--- a/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.js
+++ b/src/3rdparty/chromium/third_party/devtools-frontend/src/scripts/build/rollup.config.js
@@ -9,6 +9,8 @@
 import minifyHTML from 'rollup-plugin-minify-html-template-literals';
 import {terser} from 'rollup-plugin-terser';
 
+import process from 'process';
+
 /**
  * `path.dirname` does not include trailing slashes. If we would always
  * use `path.dirname` and then later perform comparisons on the paths that
@@ -62,6 +64,14 @@
   }
 };
 
+// nix: limit job count
+// default: $nproc - 1
+// https://github.com/TrySound/rollup-plugin-terser
+var numWorkers = 0;
+if ('NIX_BUILD_CORES' in process.env) {
+  numWorkers = parseInt(process.env.NIX_BUILD_CORES) - 1;
+}
+
 /** @type {function({configDCHECK: boolean}): import("rollup").MergedRollupOptions} */
 // eslint-disable-next-line import/no-default-export
 export default commandLineArgs => ({
@@ -81,6 +91,7 @@
       },
     }),
     terser({
+      numWorkers,
       compress: {
         pure_funcs: commandLineArgs.configDCHECK ? ['Platform.DCHECK'] : [],
       },
