--- a/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
+++ b/sources/shiboken6/ApiExtractor/clangparser/compilersupport.cpp
@@ -46,6 +46,8 @@
 #include <algorithm>
 #include <iterator>
 
+#include <iostream>
+
 namespace clang {
 
 QVersionNumber libClangVersion()
@@ -163,15 +165,40 @@
     const QByteArrayList stdErrLines = stdErr.split('\n');
     bool isIncludeDir = false;
     for (const QByteArray &line : stdErrLines) {
+
+// milahu debug
+qCInfo(lcShiboken) <<
+  "milahu 1: g++ output line = " << line.constData();
+std::cerr <<
+  "milahu 2: g++ output line = " << line.constData() << '\n';
+
+
         if (isIncludeDir) {
             if (line.startsWith(QByteArrayLiteral("End of search list"))) {
                 isIncludeDir = false;
             } else {
                 HeaderPath headerPath{line.trimmed(), HeaderType::System};
                 if (headerPath.path.endsWith(frameworkPath())) {
+// milahu debug
+qCInfo(lcShiboken) <<
+  "milahu 1: headerPath.path = " << headerPath.path.constData() << " : ends with frameworkPath = " << frameworkPath().constData();
+std::cerr <<
+  "milahu 2: headerPath.path = " << headerPath.path.constData() << " : ends with frameworkPath = " << frameworkPath().constData() << '\n';
                     headerPath.type = HeaderType::FrameworkSystem;
                     headerPath.path.truncate(headerPath.path.size() - frameworkPath().size());
+// milahu debug
+qCInfo(lcShiboken) <<
+  "headerPath.path = " << headerPath.path.constData() << " : truncated = remove frameworkPath suffix";
+std::cerr <<
+  "milahu 2: headerPath.path = " << headerPath.path.constData() << " : truncated = remove frameworkPath suffix" << '\n';
                 }
+// milahu debug
+else {
+qCInfo(lcShiboken) <<
+  "headerPath.path = " << headerPath.path.constData() << " : no frameworkPath";
+std::cerr <<
+  "milahu 2: headerPath.path = " << headerPath.path.constData() << " : no frameworkPath";
+}
                 result.append(headerPath);
             }
         } else if (line.startsWith(QByteArrayLiteral("#include <...> search starts here"))) {
