From 61ed0d5d4a9c55fe0090e4924245cf5370161189 Mon Sep 17 00:00:00 2001
From: milahu <milahu@gmail.com>
Date: Tue, 14 Feb 2023 17:21:11 +0100
Subject: [PATCH] vala: fix generated paths

---
 mesonbuild/backend/backends.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/mesonbuild/backend/backends.py b/mesonbuild/backend/backends.py
index 4a68c572e..8ebd13e88 100644
--- a/mesonbuild/backend/backends.py
+++ b/mesonbuild/backend/backends.py
@@ -808,7 +808,8 @@ class Backend:
                     rel_src = rel_src[len(build_dir) + 1:]
                 rel_src = os.path.relpath(rel_src, self.get_target_private_dir(target))
             else:
-                rel_src = os.path.basename(rel_src)
+                targetdir = self.get_target_private_dir(target)
+                rel_src = os.path.relpath(source.relative_name(), os.path.dirname(targetdir))
             # A meson- prefixed directory is reserved; hopefully no-one creates a file name with such a weird prefix.
             gen_source = 'meson-generated_' + rel_src[:-5] + '.c'
         elif source.is_built:
-- 
2.39.0

