From 79c44b9ffef9d95a52cd07a0cf4350fc767ebbaa Mon Sep 17 00:00:00 2001
From: Milan Hauth <milahu@gmail.com>
Date: Thu, 2 Feb 2023 16:48:08 +0100
Subject: [PATCH 1/2] fix codegen for unicode sources

---
 bindings/gumjs/generate-runtime.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/bindings/gumjs/generate-runtime.py b/bindings/gumjs/generate-runtime.py
index 75951b2..f19cc7e 100755
--- a/bindings/gumjs/generate-runtime.py
+++ b/bindings/gumjs/generate-runtime.py
@@ -383,6 +383,7 @@ def to_canonical_source_path(path):
 
 
 def write_bytes(data, sink):
+    "write bytes as signed integers between -128 and 127"
     sink.write("\n  ")
     line_length = 0
     offset = 0
@@ -393,6 +394,9 @@ def write_bytes(data, sink):
         if line_length >= 70:
             sink.write("\n  ")
             line_length = 0
+        # convert unsigned to signed for utf8
+        if b >= 128:
+            b -= 256
         token = str(b)
         sink.write(token)
 
-- 
2.39.0

