From 06c76221739c62bb96a7689c43fd26ea005f203b Mon Sep 17 00:00:00 2001
From: milahu <milahu@gmail.com>
Date: Thu, 16 Feb 2023 20:18:34 +0100
Subject: [PATCH] fix RAW_RIZIN_INCLUDE_DIRS + debug

---
 src/CMakeLists.txt | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 53cd202..6bff0fd 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -437,8 +437,9 @@ if(CUTTER_ENABLE_PYTHON_BINDINGS)
 
     # FIXME empty -> /build/source/src/core/CutterCommon.h:8:10: fatal: 'rz_core.h' file not found
     message("debug: Rizin_INCLUDE_DIRS = ${Rizin_INCLUDE_DIRS}")
+    message("debug: RAW_RIZIN_INCLUDE_DIRS = ${RAW_RIZIN_INCLUDE_DIRS}")
 
-    list(APPEND SHIBOKEN_INCLUDE_DIRS ${Rizin_INCLUDE_DIRS})
+    list(APPEND SHIBOKEN_INCLUDE_DIRS ${RAW_RIZIN_INCLUDE_DIRS})
     if (NOT WIN32)
         string(REPLACE ";" ":" SHIBOKEN_INCLUDE_DIRS "${SHIBOKEN_INCLUDE_DIRS}")
     endif()
@@ -561,7 +562,7 @@ if(CUTTER_ENABLE_PYTHON)
         else()
             list(APPEND BINDINGS_INCLUDE_DIRS ${Qt5Core_INCLUDE_DIRS} ${Qt5Widgets_INCLUDE_DIRS} ${Qt5Gui_INCLUDE_DIRS})
         endif()
-        list(APPEND BINDINGS_INCLUDE_DIRS ${Rizin_INCLUDE_DIRS})
+        list(APPEND BINDINGS_INCLUDE_DIRS ${RAW_RIZIN_INCLUDE_DIRS})
         list(APPEND BINDINGS_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}")
         if (NOT WIN32)
             string(REPLACE ";" ":" BINDINGS_INCLUDE_DIRS "${BINDINGS_INCLUDE_DIRS}")
-- 
2.39.0

