diff --git a/lib/rust/ensogl/component/text/src/font/msdf/build.rs b/lib/rust/ensogl/component/text/src/font/msdf/build.rs
index 06b3f4f..b197317 100644
--- a/lib/rust/ensogl/component/text/src/font/msdf/build.rs
+++ b/lib/rust/ensogl/component/text/src/font/msdf/build.rs
@@ -35,10 +35,12 @@ async fn main() -> Result {
     // If you find and implement a better way to downloading js snippets, please
     // remember to remove msdfgen_wasm.js entry from the .gitignore file.
     let mut file = ide_ci::fs::tokio::create(PACKAGE.filename).await?;
+/*
     let mut stream = ide_ci::io::web::download_reader(PACKAGE.url()?).await?;
     tokio::io::copy(&mut stream, &mut file)
         .await
         .with_context(|| format!("Failed to stream download to file {}.", PACKAGE.filename))?;
+*/
 
     // Patch downloaded msdfgen_wasm.js file.
     //
